src/browser-runner/platform-electron/sagas/server-command-handlers.js

```diff
@@ -28,6 +28,14 @@ function* createWindow({ meta: client, payload: { winId, url, width, height, sty
     }[style],
   });
 
+  win.on('close', (e) => {
+    // console.log('Window closing!!!', win.webContents);
+    // I'd like to do this only if there are no tabs left.
+    // Any ideas how I could get the number of tabs from here?
+    e.preventDefault();
+    return false;
+  });
```

Comment:
Sooooo, I think my problem with this is that electron is sending a window close event when I type `ctrl+w`, which means I need to listen for that, and call `e.preventDefault();` so that the window won't close, but sometimes (in particular, when I get the `SharedActions.events.fromFrontend.toServer.app.window.requestedClose()` action) I will want the window to close, but I'm not sure how to tell when that is the case.  (I'm going to look into it more, and maybe that's not actually a problem, but I think that might be my sticking point.)